### ISSUE TITLE
Don't set hq as the default ALPN protocol in quinn-proto

### ISSUE THREAD

**Issue by djc** [djc/quinn#32]
Leave this to higher API levels to decide.

**Comment by stammw** [comment]
Hello,

I'd like to try to solve this one !

If understand well, this ALPN parameter could be set in endpoint building code like so:

```rust
builder
    .logger(log.clone())
    .config(quinn::Config {
        max_remote_bi_streams: 64,
        alpn_protocols: vec!["hq-11"],
        ..Default::default()
}).listen();
```

or

```rust
let mut builder = quinn::Endpoint::new();
builder.set_alpn_protocols(vec!["hq-11"]);
```

These two options would let a possibility for the the user not to give any ALPN value. Is that desirable, or is it better to make this parameter mandatory?

**Comment by djc** [comment]
Hi @stammw, that would be great! I think we would want a method on `EndpointBuilder` that operates similar to `set_certificate()` and `enable_keylog()`: something that mutates the `ClientConfig` contained in the existing `Config`. @Ralith any thoughts?

**Comment by stammw** [comment]
Then TLS's ALPN parameter would be left empty when this method has never been called ?

**Comment by djc** [comment]
Yeah, I think that would be okay for some use cases, but I'll await the other reviewer.

**Comment by Ralith** [comment]
Yes, to the best of my knowledge ALPN is an optional part of the handshake.

**Comment by djc** [comment]
This has been fixed in #47. Thanks, @stammw!

**Comment by stammw** [comment]
Thank you both for all the time spent reviewing and helping !
